File: 5kyu/Product of consecutive Fib numbers/solution.js
function productFib(prod){
  let fib = [0,1];
  let x = 2;
  while(true){
    let a = fib[x-1];
    let b = fib[x-2];
    fib.push(a + b);
    if (a * b == prod){
      return [b,a, true];
    }
    else if (a * b > prod){
      return [b,a, false];          
    }
    x+=1;
  }
}

File: 5kyu/Product of consecutive Fib numbers/solution.py
def productFib(prod):
    fib = [0,1]
    x = 2
    while True:
        a = fib[x-1]
        b = fib[x-2]
        fib.append(a + b)
        if a * b == prod:
            return [b,a, True]
        elif a * b > prod:
            return [b,a, False]
        x+=1